src/estimagic/estimation/estimate_ml.py

```diff
@@ -56,20 +55,27 @@ def estimate_ml(
     The docstring is aspirational and not all options are supported yet.
 
     Args:
-        loglike (callable): Likelihood function that takes a params DataFrame (and
-            potentially other keyword arguments) and returns a dictionary that has at
-            least the entries "value" (a scalar float) and "contributions" (a 1d numpy
-            array or pandas Series) with the log likelihood contribution per individual.
-        params (pd.DataFrame): DataFrame where the "value" column contains the
-            estimated or start parameters of a likelihood model. See :ref:`params` for
-            details. If the supplied parameters are estimated parameters, set
-            optimize_options to False.
-        optimize_options (dict or False): Keyword arguments that govern the numerical
-            optimization. Valid entries are all arguments of
+        loglike (callable): Likelihood function that takes a params (and potentially
+            other keyword arguments) and returns a dictionary that has at least the
+            entries "value" (a scalar float) and "contributions" (a 1d numpy array or
+            pandas Series) with the log likelihood contribution per individual.
+        params (pytree): A pytree containing the estimated or start parameters of the
+            likelihood model. If the supplied parameters are estimated parameters, set
+            optimize_options to False. Pytrees can be a numpy array, a pandas Series, a
+            DataFrame with "value" column, a float and any kind of (nested) dictionary
+            or list containing these elements. See :ref:`params` for examples.
+        optimize_options (dict, str or False): Keyword arguments that govern the
+            numerical optimization. Valid entries are all arguments of
             :func:`~estimagic.optimization.optimize.minimize` except for criterion,
-            derivative, criterion_and_derivative and params. If you pass False as
+            criterion_and_derivative and params. If you pass False as
```

Comment:
`criterion_and_derivative` should be allowed.

Reply:
They are also passed to the function as ``loglike_and_derivative``. If we allow them in ``optimize_options`` then we need to update ``optimize_options`` instead of simply passing it to maximize.

Reply:
I misunderstood this. It is now resolved.